### PR TITLE
feat: credsStore fix in doctor

### PR DESCRIPTION
**Pull request by joshuarli**
This implements the credsStore fix in doctor as a "builtin" command since it's not repo-specific.

I hope to add some more "builtin" checks and fixes soon since I want to recommend people try `devenv doctor` before proceeding with more specific troubleshooting. One idea is the detection of docker desktop running and killing it.

https://getsentry.atlassian.net/issues/DEVINFRA-475

### CODE REVIEW

File: devenv/checks/credsStore.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+name = "credsStore"
+
+
+@checker
+def check() -> tuple[bool, str]:
+    # When Docker Desktop is opened, it inserts "desktop" as the value
+    # for credsStore. This requires docker-credential-desktop which isn't
+    # generally installed. Most people don't need to login to docker anyways.
+    with open(os.path.expanduser("~/.docker/config.json"), "rb") as f:
+        config = json.loads(f.read())
```
Comment (asottile-sentry) [inline devenv/checks/credsStore.py:20]: `json.load(f)`

File: devenv/checks/credsStore.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    return True, ""
+
+
+@fixer
+def fix() -> tuple[bool, str]:
+    try:
+        with open(os.path.expanduser("~/.docker/config.json"), "rb") as f:
+            config = json.loads(f.read())
+            if config.get("credsStore"):
+                del config["credsStore"]
```
Comment (asottile-sentry) [inline devenv/checks/credsStore.py:35]: `config.pop('credsStore', None)`

File: devenv/doctor.py [hunk 3]
```diff
@@ -106,6 +106,37 @@ def load_checks(repo: Repository, match_tags: set[str]) -> List[Check]:
...
+    Loads builtin checks.
+    Optionally filter by tags.
+    If a check doesn't have the required attributes, skip it.
+    """
+    checks: list[Check] = []
+    match_tags.add("builtin")
+
+    here = os.path.dirname(os.path.realpath(__file__))
+
+    for module_finder, module_name, _ in walk_packages((f"{here}/checks",)):
```
Comment (asottile-sentry) [inline devenv/doctor.py:120]: usually you would import the package and use `package.__path__`

File: devenv/doctor.py [hunk 4]
```diff
@@ -106,6 +106,37 @@ def load_checks(repo: Repository, match_tags: set[str]) -> List[Check]:
...
+        module_spec = module_finder.find_spec(module_name, None)
+
+        # it "should be" impossible to fail these:
+        assert module_spec is not None, module_name
+        assert module_spec.loader is not None, module_name
+
+        module = importlib.util.module_from_spec(module_spec)
+        module_spec.loader.exec_module(module)
```
Comment (asottile-sentry) [inline devenv/doctor.py:128]: shouldn't need to do the importlib dance -- just use `__import__` directly

File: devenv/checks/credsStore.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+import json
+import os
+import shutil
+
+from devenv.lib_check.types import checker
+from devenv.lib_check.types import fixer
+
+tags: set[str] = {"builtin"}
+name = "credsStore"
```
Comment (asottile-sentry) [inline devenv/checks/credsStore.py:11]: unused? also mypy wouldn't require this annotation since it can infer

File: devenv/checks/credsStore.py [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+    return True, ""
+
+
+@fixer
+def fix() -> tuple[bool, str]:
+    try:
+        with open(os.path.expanduser("~/.docker/config.json"), "rb") as f:
+            config = json.load(f)
+            config.pop("credsStore", None)
```
Comment (asottile-sentry) [inline devenv/checks/credsStore.py:34]: it doesn't actually matter but I would dedent this and put a newline before it (no reason to hold a file open longer than it needs to be)

File: devenv/doctor.py [hunk 7]
```diff
@@ -106,6 +106,28 @@ def load_checks(repo: Repository, match_tags: set[str]) -> List[Check]:
...
+    If a check doesn't have the required attributes, skip it.
+    """
+    checks: list[Check] = []
+    match_tags.add("builtin")
+
+    for _, module_name, _ in walk_packages(devenv.checks.__path__):
+        module = __import__(f"devenv.checks.{module_name}", fromlist=["_trash"])
+        try:
+            check = Check(module)
+        except AssertionError as e:
```
Comment (asottile-sentry) [inline devenv/doctor.py:122]: `except AssertionError` is usually a smell -- this should be ValueError or TypeError probably?

File: devenv/checks/credsStore.py (hunk 5 above)
Comment (joshuarli) [reply devenv/checks/credsStore.py:11]: it's used during construction of `doctor.Check`s

File: devenv/doctor.py [hunk 8]
```diff
@@ -119,7 +119,7 @@ def load_builtin_checks(match_tags: set[str]) -> List[Check]:
         module = __import__(f"devenv.checks.{module_name}", fromlist=["_trash"])
         try:
             check = Check(module)
-        except AssertionError as e:
+        except TypeError as e:
```
Comment (asottile-sentry) [inline devenv/doctor.py:122]: eheh now that I see the code I actually think `ValueError` is more appropriate

**Review by asottile-sentry — APPROVED** [review]

**Pull request merged**